switch.go

```diff
@@ -340,22 +307,22 @@ func (sw *Switch) dialSeed(addr *NetAddress) {
 	}
 }
 
-func (sw *Switch) DialPeerWithAddress(addr *NetAddress) (*Peer, error) {
-	log.Info("Dialing address", "address", addr)
+func (sw *Switch) DialPeerWithAddress(addr *NetAddress, persistent bool) (*Peer, error) {
 	sw.dialing.Set(addr.IP.String(), addr)
-	conn, err := addr.DialTimeout(time.Duration(
-		sw.config.GetInt(configKeyDialTimeoutSeconds)) * time.Second)
-	sw.dialing.Delete(addr.IP.String())
+	peer, err := newPeer(addr, sw.reactorsByCh, sw.chDescs, sw.StopPeerForError, sw.config, sw.nodePrivKey)
+	if persistent {
+		peer.MakePersistent()
+	}
 	if err != nil {
-		log.Info("Failed dialing address", "address", addr, "error", err)
+		log.Info("Failed dialing peer", "address", addr, "error", err)
+		peer.CloseConn()
 		return nil, err
 	}
-	if sw.config.GetBool(configFuzzEnable) {
-		conn = FuzzConn(sw.config, conn)
-	}
-	peer, err := sw.AddPeerWithConnection(conn, true)
+	err = sw.AddPeer(peer)
+	sw.dialing.Delete(addr.IP.String())
```

Comment:
Actually, maybe we should pass the `dialing Cmap` to `func dial` ... `Redial` doesn't seem to set the peer in `dialing`